### ISSUE TITLE
Interactive Markers disappeared in Motion Planning/Rviz in recent versions

### ISSUE THREAD

**Issue by skohlbr** [moveit/moveit_ros#225]
I generated a robot config using the setup assistant a few weeks ago. Running the demo.launch today for testing changes I had to perform to comply with the new kinematics_base interface (I use ikfast based custom plugins), I noticed that Interactive Markers are no more appearing in rviz. I removed/re-added the display to no avail. Any suggestions what changes I have to perform or what the issue is would be appreciated.

**Comment by isucan** [comment]
Does the plugin connect successfully to the motion planner? i.e., can you still plan motions (even if with the same start & goal locations)? 
Try setting the interactive marker size to something above 0, in the rviz display. 
If that does not work, enable debugging (run rqt_logger_levels) for move_group and check what debug messages say when you change groups in the rviz plugin (and perhaps attach to this ticket)

**Comment by skohlbr** [comment]
Yes, it is still possible to connect to the planner. Setting random start and goal configurations, planning, and visualizing movement all work. Setting IM size to be greater than 0 had no effect (sounds like a good idea though).

The move_group output looks pretty normal. Switching between left and arm groups results in the following INFOs (DEBUG was enabled for nearly every part of move_group with the exception of roscpp etc.):

<pre>
[ INFO] [1366059118.157260920]: Ready to take MoveGroup commands for group l_arm_group.
[ INFO] [1366059134.126274352]: Constructing new MoveGroup connection for group 'r_arm_group'
[ INFO] [1366059134.127816913]: Waiting for MoveGroup action server (move_group)...
[ INFO] [1366059134.149205950]: Connected to 'move_group'
[ INFO] [1366059134.149795577]: Waiting for MoveGroup action server (pickup)...
[ INFO] [1366059134.170303782]: Connected to 'pickup'
[ INFO] [1366059134.170610478]: Waiting for MoveGroup action server (place)...
[ INFO] [1366059134.190885052]: Connected to 'place'
[ INFO] [1366059134.191165195]: Ready to take MoveGroup commands for group r_arm_group.
[ INFO] [1366059172.896077423]: Constructing new MoveGroup connection for group 'l_arm_group'
[ INFO] [1366059172.896685288]: Waiting for MoveGroup action server (move_group)...
[ INFO] [1366059172.916817937]: Connected to 'move_group'
[ INFO] [1366059172.917024417]: Waiting for MoveGroup action server (pickup)...
[ INFO] [1366059172.937251803]: Connected to 'pickup'
[ INFO] [1366059172.937619288]: Waiting for MoveGroup action server (place)...
[ INFO] [1366059172.958851623]: Connected to 'place'
[ INFO] [1366059172.959117563]: Ready to take MoveGroup commands for group l_arm_group.
</pre>

**Comment by isucan** [comment]
This is a bit odd. I would expect at least some warnings about no markers found. Is it possible for you to e-mail me your config + urdf + meshes?
Things to re-check: Do you have an end effector defined? (not needed with recent versions) Is your group a chain? Is TF being published? Is there an IK solver associated to the group? Can you confirm the IK solver gets initialized successfully ?

**Comment by isucan** [comment]
can you check again with the most recent version?

**Comment by skohlbr** [comment]
Just checked the PR2, it has IMs. Will try most recent version now. I saw KinematicsBase has changed again, so will have to fix our own plugins also.

**Comment by skohlbr** [comment]
Works with most recent version. I also noticed I had a non-existing fixed frame set in my rviz, so the IMs probably couldn't be transformed properly. Being used to how rviz normally works (e.g. not visualizing the robot when wrong frame selected), I failed to notice the error. It might make sense to show a more prominent warning somewhere when this is the case.

Thus, case closed from my POV. Thanks again for the quick replies.
